### DISCUSSION TITLE
Read file_sd_config label as _param_auth

### DISCUSSION THREAD

**Discussion by j4mbob** [prometheus/snmp_exporter#928] [Q&A] [answered]
Hi, I've been following the recent discussion about the addition of split auth. I have json file with all my switches defined in it with a label which contains the auth profile to use:

```
[
  {
    "targets": [
      "x.x.x.x"
    ],
   "labels" : { "community":"switch1"}

  }
]
```

in my prom yaml file Im loading this via:

 - job_name: 'snmp'
          file_sd_configs:
            - files:
              - 'switches.json'

I want to set __param,_auth to use  the community label embedded in switches.json via relabeling - i tried :

            - source_labels: community
              target_label: __param_auth

and various variations but I just get an unmarshalling error: : cannot unmarshal !!str `community` into model.LabelNames

which presumably means it cant find the label.

This should be possible now right?

as a side note its a shame you cant directly pass the community string. it would be much less of a pain (I have 1000s of devices I want to monitor all of the same type and all with a unique community string, but i understand the reason for not allowing it!)

thanks for any assistance

**Comment by SuperQ** [comment, marked as answer]
```yaml
- source_labels: [community]
  target_label: __param_auth
```

**Reply by j4mbob** [reply]
awesome, out of all the variations i tried... that wasnt one!

havent quite grasped what the []'s represent in labels yet (only started using prom last week)

 thank you !

**Reply by SuperQ** [reply]
This is not related to labels, this is yaml syntax. `source_labelS`, it's plural, it implies it's a list.

You can also write it this way:
```yaml
- source_labels:
    - community
  target_label: __param_auth
```

**Reply by j4mbob** [reply]
ah yes, of course, makes perfect sense. thanks

I noticed this earlier - https://github.com/prometheus/snmp_exporter/blob/main/testdata/snmp-auth.yml

can the auth config now be split off from the main snmp.yml file?

how is it loaded?

sorry for all the questions!

**Reply by SuperQ** [reply]
> can the auth config now be split off from the main snmp.yml file?

No, not yet. Everything must still be merged into one config file.

There is an open issue to allow loading multiple files. Merging multiple config files is tricky.
